src/core/mac/mac.cpp

```diff
@@ -817,8 +823,11 @@ void Mac::SentFrame(bool aAcked)
             ExitNow();
         }
 
-        sendFrame.GetDstAddr(destination);
-        mMle.RemoveNeighbor(destination);
+        break;
+
+    default:
+        assert(false);
```

Comment:
If other errors are allowed, then we need to handle them properly.  It's not yet obvious to me what to do in the default case (e.g. try transmitting again, pass the error up to the higher layer, etc.).